### PR TITLE
#1593165508 Refactor Authors Haven codebase to ES6

### PR DESCRIPTION
**Pull request by mppdg**
#### What does this PR do?
- This PR converts the existing codebase from ES5 to ES6.

#### Description of Task to be completed?
- install and configure babel to transpile the codebase
- refactor codes in each file to ES6 in the existing codebase
- rebase and refactor new merges to ES6

#### How should this be manually tested?
- `npm start`

#### Any background context you want to provide?
- The refactoring includes existing codebase and PRs that have been merged.  We advise that others write their code in ES6 style going forward.

#### What are the relevant pivotal tracker stories?
- #1593165508

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -70,16 +67,17 @@ if (!isProduction) {
 
 // production error handler
 // no stacktraces leaked to user
-app.use(function (err, req, res, next) {
+app.use((err, req, res) => {
```
Comment (megame24) [inline index.js]: you omitted the `next` parameter in the refactored middleware.

Comment (JCanaks) [reply index.js]: Apart from Resolving the hound errors and including the next parameter as stated in the comment above. I believe the PR is good to go.

File: routes/api/users.js [hunk 2]
```diff
@@ -1,87 +1,87 @@
...
-    })(req, res, next);
+    return res.status(422).json(info);
+  })(req, res, next);
+  return undefined;
 });
 
-router.post("/users", function(req, res, next) {
-    const username = req.body.user.username,
+router.post('/users', (req, res, next) => {
+  const username = req.body.user.username,
```
Comment (megame24) [inline routes/api/users.js]: You should take care of this destructuring prompt for `const username = req.body.user.username`

**Review by megame24 — APPROVED** [review]

**Review by rhenshaw56 — CHANGES_REQUESTED** [review]
you still have some hound violations, please fix those. there should be no violations on this pr

File: index.js [hunk 3]
```diff
@@ -1,90 +1,83 @@
...
 });
 
-/// error handlers
-
+// error handlers
 // development error handler
 // will print stacktrace
 if (!isProduction) {
-  app.use(function (err, req, res, next) {
+  app.use((err, req, res, next) => { //eslint-disable-line no-unused-vars
```
Comment (tutugodfrey) [inline index.js]: Please add the space after forword slashs `//` when commenting so hound won't cry

File: config/passport.js [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
+        const validPassword = (thePassword) => {
+          const hash = crypto
+            .pbkdf2Sync(thePassword, user.salt, 10000, 512, 'sha512')
+            .toString('hex');
+          return user.hash === hash;
+        };
+        if (!user || !validPassword(password)) {
+          return done(null, false, {
+            errors: { 'email or password': 'is invalid' },
+          });
```
Comment (tutugodfrey) [inline config/passport.js:25]: Can your try to clear out the indentation errors.

**Review by chidioguejiofor — COMMENTED** [review]
Looks good to me once the hound comments  are checked

**Review by chidioguejiofor — APPROVED** [review]

**Review by tutugodfrey — APPROVED** [review]

**Review by JCanaks — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

**Pull request merged**